alchemist/alchemist-swingui/src/main/java/it/unibo/alchemist/boundary/gui/effects/DrawLayers.java

```diff
@@ -0,0 +1,292 @@
+    private Color colorCache = Color.BLUE;
+    @Nullable
+    @SuppressFBWarnings("SE_TRANSIENT_FIELD_NOT_RESTORED")
+    private transient Molecule molecule;
+    @Nullable
+    @SuppressFBWarnings("SE_TRANSIENT_FIELD_NOT_RESTORED")
+    private transient Object molStringCached;
+    @SuppressFBWarnings(value = "SE_TRANSIENT_FIELD_NOT_RESTORED", justification = "If null, it gets reinitialized anyway if needed")
+    private transient CollectionWithCurrentElement<String> prevIncarnation;
+    private transient Incarnation<?, ?> incarnation;
+    private transient Optional<Node> markerNode = Optional.empty();
+
+    /**
+     */
+    public DrawLayers() {
+        final Set<String> availableIncarnations = SupportedIncarnations.getAvailableIncarnations();
+        if (availableIncarnations.isEmpty()) {
+            throw new IllegalStateException(getClass().getSimpleName() + " can't work if no incarnation is available.");
+        }
+        curIncarnation = new ImmutableCollectionWithCurrentElement<>(availableIncarnations, availableIncarnations.stream().findAny().get());
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @SuppressWarnings({"PMD.CompareObjectsWithEquals", "unchecked"})
+    @SuppressFBWarnings("ES_COMPARING_STRINGS_WITH_EQ")
+    @Override
+    public <T, P extends Position2D<P>> void apply(final Graphics2D g, final Node<T> n, final Environment<T, P> env, final IWormhole2D<P> wormhole) {
+        // if marker node is no longer in the environment or it is no longer displayed, we need to change it
+        if (markerNode.isPresent() && (!env.getNodes().contains(markerNode.get())
+                                        || !wormhole.isInsideView(wormhole.getViewPoint(env.getPosition((Node<T>) markerNode.get()))))) {
+            markerNode = Optional.empty();
+        }
+        if (markerNode.isEmpty()) {
+            markerNode = Optional.of(n);
+        }
+        if (markerNode.get() == n) { // at this point markerNode.isPresent() is always true, so we directly get it
+            if (layerFilter && (incarnation == null || curIncarnation != prevIncarnation || molString != molStringCached)) {
+                molStringCached = molString;
+                prevIncarnation = curIncarnation;
+                incarnation = SupportedIncarnations.get(curIncarnation.getCurrent())
+                        .orElseThrow(() -> new IllegalStateException(curIncarnation.getCurrent() + " is not a valid incarnation."));
+                /*
+                 * Process in a separate thread: if it fails, does not kill EDT.
+                 */
+                final Thread th = new Thread(() -> molecule = incarnation.createMolecule(molString));
+                th.start();
+                try {
+                    th.join();
+                } catch (final InterruptedException e) {
+                    L.error("Bug.", e);
+                }
+            }
+
```

Comment:
No empty lines. If there is a separator, then write a comment. Otherwise, compact the code.

Reply:
Ok, I removed all the empty lines (or added a comment). I will resolve that conversations too